pybossa/view/projects.py

```diff
@@ -1697,6 +1696,28 @@ def respond():
 
         return handle_content_type(data)
 
+    def get_users_fullname(page_tasks, get_users_func, result_field):
+        user_info = {}
+        # Enumerate all tasks in the page.
+        for task in page_tasks:
+            users = []
+            # Retrieve the list of user ids to obtain names for (from either the task or task_run, selectable by calling func).
+            users_list = get_users_func(task)
+            # For each user id, load the fullname.
+            for user_id in users_list:
+                try:
+                    # show fullname for users
+                    user_id = int(user_id)
+                    if not user_info.get(user_id):
+                        # Load from cache.
+                        user_info[user_id] = cached_users.get_user_by_id(user_id).fullname
+                    users.append(user_info[user_id])
+                except ValueError:
+                    # Error locating user id.
+                    current_app.logger.info("User does not have valid user id in get_users_fullname(): %s", user_id)
+            task[result_field] = ", ".join(users)
+        return user_info
```

Comment:
Looks like the return statement is not necessary.

Reply:
The function could certainly return None. Returning a value from user_info was intended to make future unit testing efforts easier. If there is a specific preference to return a specific value, we could adjust.